go/obscuronode/enclave/state.go

```diff
@@ -110,16 +110,28 @@ func updateState(b *types.Block, s Storage, blockResolver BlockResolver) *blockS
 
 	rollups := extractRollups(b, blockResolver)
 
-	// The genesis rollup is part of the canonical chain and will be included in an L1 block by the first Aggregator.
-	if len(rollups) == 1 && rollups[0].Hash() == GenesisRollup.Hash() {
-		bs := blockState{
-			block:          b,
-			head:           &GenesisRollup,
-			state:          emptyState(),
-			foundNewRollup: true,
+	genesisRollup := s.FetchRollupGenesis()
+	if len(rollups) == 1 {
+		rol := rollups[0]
+
+		// the incoming block might hold the genesis rollup
+		// todo change this to an hardcoded hash on testnet/mainnet
```

Comment:
basically this if assumes that the genesis rollup hash will be passed in as a config.
Makes sense